src/wh_crypto.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2024 wolfSSL Inc.
+ *
+ * This file is part of wolfHSM.
+ *
+ * wolfHSM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfHSM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with wolfHSM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+ * src/wh_crypto.c
+ *
+ * Common crypto functions for both the client and server
+ *
+ */
+
+/* Pick up compile-time configuration */
+#include "wolfhsm/wh_settings.h"
+
+#ifndef WOLFHSM_CFG_NO_CRYPTO
+
+/* System libraries */
+#include <stdint.h>
+#include <stddef.h>  /* For NULL */
+
+#include "wolfssl/wolfcrypt/settings.h"
+#include "wolfssl/wolfcrypt/types.h"
+#include "wolfssl/wolfcrypt/error-crypt.h"
+#include "wolfssl/wolfcrypt/asn.h"
+#include "wolfssl/wolfcrypt/ecc.h"
+
+#include "wolfhsm/wh_error.h"
+#include "wolfhsm/wh_utils.h"
+
+#include "wolfhsm/wh_crypto.h"
+
+#ifdef HAVE_ECC
+int wh_Crypto_EccSerializeKey(ecc_key* key, uint16_t max_size,
```

Comment:
"raw" is used in asn.c which means ASCII hex, a shock to me as well.  "unsigned" means an ASN.1 DER binary byte sequence.  I had initially called these functions with Der in the name, but once we add in the symmetric keys, there's no need for DER encoding at all.  For asymmetric sequences, the DER also includes the curve id and key size, both of which I think are encoded in the wolfBoot metadata rather than being attached to the concatenated binary values.  These functions were specifically intended to ensure the representation of key material was consistent between the client and the server.  Currently, the server also chooses to use this same format to commit the key to NVM, but that could be changed without impacting the client.

I think I'll add Der back into the name for the asymmetric ones so that it is clear that it could be parsed as valid ASN.1.  If you think there's value with adding a wolfBoot flavor here, let's call it SerializeKeyWolfBoot, or something that doesn't overload "raw", "binary", or "unsigned" again.